mne/digitization/_utils.py

```diff
@@ -0,0 +1,395 @@
+                             'instead of %s' % (extra_points.shape,))
+        for idx, point in enumerate(extra_points):
+            dig.append({'r': point, 'ident': idx + 1,
+                        'kind': FIFF.FIFFV_POINT_EXTRA,
+                        'coord_frame': FIFF.FIFFV_COORD_HEAD})
+    if dig_ch_pos is not None:
+        keys = sorted(dig_ch_pos.keys())
+        try:  # use the last 3 as int if possible (e.g., EEG001->1)
+            idents = [int(key[-3:]) for key in keys]
+        except ValueError:  # and if any conversion fails, simply use arange
+            idents = np.arange(1, len(keys) + 1)
+        for key, ident in zip(keys, idents):
+            dig.append({'r': dig_ch_pos[key], 'ident': ident,
+                        'kind': FIFF.FIFFV_POINT_EEG,
+                        'coord_frame': FIFF.FIFFV_COORD_HEAD})
+
+    return _format_dig_points(dig)
+
+
+def _call_make_dig_points(nasion, lpa, rpa, hpi, extra, convert=True):
+    if convert:
+        neuromag_trans = get_ras_to_neuromag_trans(nasion, lpa, rpa)
+        nasion = apply_trans(neuromag_trans, nasion)
+        lpa = apply_trans(neuromag_trans, lpa)
+        rpa = apply_trans(neuromag_trans, rpa)
+
+        if hpi is not None:
+            hpi = apply_trans(neuromag_trans, hpi)
+
+        extra = apply_trans(neuromag_trans, extra).astype(np.float32)
+    else:
+        neuromag_trans = None
+
+    ctf_head_t = Transform(fro='ctf_head', to='head', trans=neuromag_trans)
+
+    info_dig = _make_dig_points(nasion=nasion,
+                                lpa=lpa,
+                                rpa=rpa,
+                                hpi=hpi,
+                                extra_points=extra)
+
+    return info_dig, ctf_head_t
+
+
+##############################################################################
+# From mne.io.kit
+def _set_dig_kit(mrk, elp, hsp):
+    """Add landmark points and head shape data to the KIT instance.
+
+    Digitizer data (elp and hsp) are represented in [mm] in the Polhemus
+    ALS coordinate system. This is converted to [m].
+
+    Parameters
+    ----------
+    mrk : None | str | array_like, shape = (5, 3)
+        Marker points representing the location of the marker coils with
+        respect to the MEG Sensors, or path to a marker file.
+    elp : None | str | array_like, shape = (8, 3)
+        Digitizer points representing the location of the fiducials and the
+        marker coils with respect to the digitized head shape, or path to a
+        file containing these points.
+    hsp : None | str | array, shape = (n_points, 3)
+        Digitizer head shape points, or path to head shape file. If more
+        than 10`000 points are in the head shape, they are automatically
+        decimated.
+
+    Returns
+    -------
+    dig_points : list
+        List of digitizer points for info['dig'].
+    dev_head_t : dict
+        A dictionary describe the device-head transformation.
```

Comment:
do we have a template docstring for this?

Reply:
No I'll make a PR.

Reply:
￼I'll push them in #6385